pkg/scheduler/objects/queue.go

```diff
@@ -1220,6 +1220,50 @@ func (sq *Queue) GetMaxResource() *resources.Resource {
 	return sq.internalGetMax(limit)
 }
 
+// GetFairMaxResource computes the fair max resources for a given queue.
+// Starting with the root, descend down to the target queue allowing children to override Resource values .
+// If the root includes an explicit 0 value for a Resource, do not include it in the accumulator and treat it as missing.
+// If no children provide a maximum capacity override, the resulting value will be the value found on the Root.
+// It is useful for fair-scheduling to allow a ratio to be produced representing the rough utilization % of a given queue.
+func (sq *Queue) GetFairMaxResource() *resources.Resource {
+	var limit *resources.Resource
+	if sq.parent == nil {
+		cleaned := resources.NewResource()
+		if sq.maxResource == nil {
+			return cleaned
+		}
+
+		for k, v := range sq.maxResource.Resources {
+			if v != 0 {
+				cleaned.Resources[k] = v
+			}
+		}
```

Comment:
If we wait a little bit for #943, then we'll be able to call `cleaned.Prune()`.

IMO this is the most readable approach:
```
if sq.maxResource == nil {
  return resources.NewResource()
}

return sq.maxResources.Clone().Prune()
```

Or perhaps we shouldn't even allow "0" resources in max in the first place and we just clone it...

Reply:
0 resources need to be allowed as they may be explicitly set on queues. Only on the root queue do we ignore zero.

Reply:
Thx, good to know.